src/microbe_stage/MicrobeAI.cs

```diff
@@ -9,385 +10,179 @@
             {
-                if (!predator.Dead)
+                if (chunk.ContainedCompounds.Compounds.Any(x => microbe.Compounds.IsUseful(x.Key)))
                 {
-                    if ((microbe.Translation - predator.Translation).LengthSquared() <=
-                        (2000 + ((predator.HexCount * 8.0f) * 2)))
+                    chunkList.Add(chunk);
+
+                    if (chosenChunk == null ||
+                        (chosenChunk.Translation - microbe.Translation).LengthSquared() >
+                        (chunk.Translation - microbe.Translation).LengthSquared())
                     {
-                        if (lifeState != LifeState.FLEEING_STATE)
-                        {
-                            // Reset target position for faster fleeing
-                            hasTargetPosition = false;
-                        }
-
-                        boredom = 0;
-                        lifeState = LifeState.FLEEING_STATE;
+                        chosenChunk = chunk;
                     }
                 }
-                else
-                {
-                    predator = null;
-                }
-            }
-            catch (ObjectDisposedException)
-            {
-                // Our predator might be already disposed
-                predator = null;
             }
         }
-    }
-
-    /// <summary>
-    /// Gets the nearest chunk. And builds a list on chunkList
-    /// </summary>
-    /// <returns>The nearest chunk item.</returns>
-    /// <param name="allChunks">All chunks the AI knows of.</param>
-    private FloatingChunk GetNearestChunkItem(List<FloatingChunk> allChunks)
-    {
-        FloatingChunk chosenChunk = null;
 
-        Vector3 testPosition = new Vector3(0, 0, 0);
-        bool setPosition = true;
-
-        // Retrieve nearest potential chunk
-        foreach (var chunk in allChunks)
+        // Don't bother with chunks when there's a lot of microbes to compete with
+        if (chosenChunk != null)
         {
-            if (chunk == previousChunk)
-                continue;
-
-            if ((SpeciesOpportunism == Constants.MAX_SPECIES_OPPORTUNISM) ||
-                ((microbe.EngulfSize * (SpeciesOpportunism / Constants.OPPORTUNISM_DIVISOR)) >
-                    chunk.Size))
+            var numRivals = 0;
+            foreach (var rival in allMicrobes)
             {
-                chunkList.Add(chunk);
-                var thisPosition = chunk.Translation;
-
-                if (setPosition)
+                if (rival != microbe)
                 {
-                    testPosition = thisPosition;
-                    setPosition = false;
-                    chosenChunk = chunk;
+                    var rivalDistance = (rival.Translation - chosenChunk.Translation).LengthSquared();
+                    if (rivalDistance < 500.0f &&
+                        rivalDistance < (microbe.Translation - chosenChunk.Translation).LengthSquared())
```

Comment:
`microbe.Translation - chosenChunk.Translation` should be precomputed outside the loop just once